tests/e2escenarios/e2e_test.go

```diff
@@ -259,6 +259,230 @@ var _ = Describe("E2E Test", func() {
+
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+
+			//add binding information (binding as file)
+			helper.Cmd("odo", "add", "binding", "--name", bindingName, "--service", "cluster-sample").ShouldPass()
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js", "from updated Node.js")
+			Expect(err).ToNot(HaveOccurred())
+
+			_, _, _, err = devSession.WaitSync()
+			Expect(err).ToNot(HaveOccurred())
+
+			// "should update the changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from updated Node.js Starter Application!")
+
+			// check odo describe to check for env
+			stdout := helper.Cmd("odo", "describe", "binding", "-o", "json").ShouldPass().Out()
+			Expect(helper.IsJSON(stdout)).To(BeTrue())
+			helper.JsonPathContentIs(stdout, "0.name", bindingName)
+			helper.JsonPathContentIs(stdout, "0.spec.application.kind", "Deployment")
+			helper.JsonPathContentIs(stdout, "0.spec.application.name", componentName+"-app")
+			helper.JsonPathContentIs(stdout, "0.spec.application.apiVersion", "apps/v1")
+			helper.JsonPathContentIs(stdout, "0.spec.services.0.apiVersion", "postgresql.k8s.enterprisedb.io/v1")
+			helper.JsonPathContentIs(stdout, "0.spec.services.0.kind", "Cluster")
+			helper.JsonPathContentIs(stdout, "0.spec.services.0.name", "cluster-sample")
+			helper.JsonPathContentIs(stdout, "0.spec.detectBindingResources", "true")
+			helper.JsonPathContentIs(stdout, "0.spec.bindAsFiles", "true")
+			helper.JsonPathContentContain(stdout, "0.status.bindingFiles", fmt.Sprintf("%q", "${SERVICE_BINDING_ROOT}/"+bindingName+"/password"))
+
+			// check if secrets are mounted in
+			podName := commonVar.CliRunner.GetRunningPodNameByComponent(componentName, commonVar.Project)
+			stdout = commonVar.CliRunner.Exec(podName, commonVar.Project, "bash", "-c", "ls", "'${SERVICE_BINDING_ROOT}/"+bindingName+"/password'")
+			Expect(stdout).Should(Not(BeEmpty()))
+
+			// "changes are made made to the applications"
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from updated Node.js", "from Node.js app v2")
+			_, _, _, err = devSession.WaitSync()
+			Expect(err).ToNot(HaveOccurred())
+
+			// "should deploy new changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from Node.js app v2 Starter Application!")
+
+			// "running odo list"
+			stdout = helper.Cmd("odo", "list").ShouldPass().Out()
+			helper.MatchAllInOutput(stdout, []string{componentName, "nodejs", "Dev"})
+
+			// remove bindings and check devfile to not contain binding info
+			helper.Cmd("odo", "remove", "binding").ShouldPass()
```

Comment:
`odo remove binding` requires a binding name:
```
❯ odo remove binding
 ✗  you must specify the service binding name with --name flag
```

Also, since `odo remove binding` would edit the Devfile, the running Dev session should normally restart the container and the application. So I think the end-to-end test could go further and check the application (with a project that consumes the binding information, the application would not be reachable correctly).